### ISSUE TITLE
Banner ads cause java.lang.RuntimeException on Flutter 3.16.0

### ISSUE THREAD

**Issue by gkovac42** [AppLovin/AppLovin-MAX-Flutter#147]
### MAX Plugin Version

3.3.0

### Flutter Version

3.16.0

### Device/Platform Info

Android

### Current Behavior

Ever since I updated to Flutter 3.16.0 the app crashes when returning from background if there are Banner ads on the screen. I get this error in the Play Console ANR section:

```
Exception java.lang.RuntimeException: Unable to resume activity {com.grimdev.grimquest0/com.grimdev.grimquest0.MainActivity}: java.lang.IllegalStateException: Failed to load new ad - this instance is already destroyed for ad unit ID: 80fea5540984db13
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4820)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4853)
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:54)
  at android.app.servertransaction.ActivityTransactionItem.execute (ActivityTransactionItem.java:45)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2218)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at android.app.ActivityThread.main (ActivityThread.java:8005)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:548)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1039)
Caused by java.lang.IllegalStateException: Failed to load new ad - this instance is already destroyed for ad unit ID: 80fea5540984db13
  at com.applovin.impl.mediation.ads.MaxAdViewImpl.a (SourceFile:252)
  at com.applovin.impl.mediation.ads.MaxAdViewImpl.loadAd (SourceFile:234)
  at com.applovin.impl.mediation.ads.MaxAdViewImpl.onAdRefresh (SourceFile:649)
  at com.applovin.impl.sdk.e.AA (SourceFile:297)
  at com.applovin.impl.sdk.e.onReceive (SourceFile:228)
  at com.applovin.impl.sdk.AppLovinBroadcastManager.sendBroadcastSync (SourceFile:232)
  at com.applovin.impl.sdk.SessionTracker.Ft (SourceFile:219)
  at com.applovin.impl.sdk.SessionTracker.AA (SourceFile:159)
  at com.applovin.impl.sdk.SessionTracker.a
  at com.applovin.impl.sdk.SessionTracker$1.onActivityResumed (SourceFile:69)
  at android.app.Application.dispatchActivityResumed (Application.java:416)
  at android.app.Activity.dispatchActivityResumed (Activity.java:1405)
  at android.app.Activity.onResume (Activity.java:1958)
  at io.flutter.embedding.android.FlutterActivity.onResume (FlutterActivity.java:795)
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1477)
  at android.app.Activity.performResume (Activity.java:8211)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4810)
```

Flutter Doctor:

```
[√] Flutter (Channel stable, 3.16.0, on Microsoft Windows [Version 10.0.19045.3693], locale hr-HR)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.3.4)
[√] Android Studio (version 2023.1)
[√] Connected device (4 available)
[√] Network resources
```

### How to Reproduce

Update to 3.16.0 and run banner ads. Switch from background to foreground.

**Comment by AviNLeung** [comment]
Thanks for bringing this up. It looks like you might be trying to load a banner ad that has already been destroyed. Can you make sure the destroy method is not called before the load method and let us know if that resolves the issue?

**Comment by AviNLeung** [comment]
Feel free to use this similar issue on Android for reference: https://github.com/AppLovin/AppLovin-MAX-SDK-Android/issues/583

**Comment by gkovac42** [comment]
I'm not calling any ad related methods on lifecycle changes like in the reference issue. I only initiate the banner ad in the init method of the main screen, as per documentation.

**Comment by AviNLeung** [comment]
Thanks for clarifying this. Just to confirm, are there any places where you are calling the `destroyBanner` or other `destroy`/`dispose` methods?

**Comment by AviNLeung** [comment]
In sdk 12.0.0 we started notifying publishers when they try to load ads that have been destroyed, because it is a missed opportunity to display an ad. 

We will investiate the issue you have brought up with us further.  Could you provide any logs from the crash? Could you also provide us a way to display banners in your app. We tried to display banners to reproduce the issue, and were unable to do so in your app.

**Comment by kientux** [comment]
This issue is fixed in SDK v12.1.0, please release a new version of flutter sdk with upgraded android sdk @AviNLeung

**Comment by AviNLeung** [comment]
No worries, thanks for bringing the original issue to our attention! @kientux
